examples/Metadata Extraction/Metadata-extraction.ps1

```diff
@@ -0,0 +1,298 @@
+# Prompt for params if some are missing
+if (-not$FolderID)
+{
+    Write-Log "Please enter the ID of the folder to extract metadata from:" -output true -color Yellow
+    $FolderID = Read-Host
+}
+
+if (-not$UserId)
+{
+    Write-Log "Please specify the user ID  to pull metadata as a specific user" -output true -color Yellow
+    Write-Log "Press Enter if you want to run script as the current user." -output true -color Yellow
+    $UserId = Read-Host "User ID"
+}
+
+Function Start-Metadata-Extraction {
+
+    Write-Log "Pulling data from Folder ID: $FolderID" -output true -color Green
+
+    try {
+        #Pull all File ID values from Folder ID
+        If (!$UserId) {
+            #Run as default user
+            Write-Log "No user ID specified. Using current user." -output true -color Yellow
+            $EntriesResp = (box folders:items $FolderID --json 2>&1)
+        }
+        Else {
+            Write-Log "Extracting metadata as user ID: $UserId" -output true -color Yellow
+            #Run with as-user header with inputted User ID
+            $EntriesResp = (box folders:items $FolderID --as-user=$UserId --json 2>&1)
+
+        }
+
+        $Entries = $EntriesResp | ConvertFrom-Json
+
+    }
+    catch {
+        Write-Log "Could not get the folder items. See error log for details." -errorMessage $EntriesResp -output True -color Red
+        break
+    }
+
+    Write-Log "Output $($Entries | Out-String)"
+
+    ForEach ($Item in $Entries) {
+        $ItemID = $Item.id
+        Write-Log "Reading Item ID: $ItemID" -output true -color Green
+
+        #Pull Metadata values from Folder ID's
+        try {
+            If (!$UserId) {
+                #Run as default user (service account)
+                If ($Item.type -eq 'file') {
+                $MetadataResp = (box files:metadata $ItemID --json 2>&1)
+                }
+                elseif ($Item.type -eq 'folder') {
+                    $MetadataResp = (box folders:metadata $ItemID --json 2>&1)
+                }
+            }
+            Else {
+                #Run with as-user header with inputted User ID
+                If ($Item.type -eq 'file') {
+                    $MetadataResp = (box files:metadata $ItemID --as-user=$UserId --json 2>&1)
+                }
+                elseif ($Item.type -eq 'folder') {
+                    $MetadataResp = (box folders:metadata $ItemID --as-user=$UserId --json 2>&1)
+                }
+
+            }
+
+            $Metadata = $MetadataResp | ConvertFrom-Json
+
+        }
+        catch {
+            Write-Log "Could not get the metadata for item. See error log for details." -errorMessage $MetadataResp -output True -color Red
+            continue
+        }
+
+
+        #Loop through each metadata entry to add additional folder info & separate according to metadata template
+        ForEach ($MetadataValue in $Metadata) {
+            #Append File Info values: Name, File Id, & Type
```

Comment:
Do you mean to use `$Item` instead of `$MetadataValue`?

Reply:
I mean the name of the column only

Reply:
I believe `id` is taken from API response directly and we don't modify it.

Reply:
I mean the *column name* in the csv file, it should be `Object ID` or `Item ID` because it contains folder and file, not only file so the name `File Id` is not correct.

For example:

```
#Append File Info values: Name, Object Id, & Type
...
$MetadataValue | Add-Member -NotePropertyName "Object Id" -NotePropertyValue $Item.id;
```

Reply:
got you, updated the branch.